syntax = 'proto3';
import "google/protobuf/timestamp.proto";
message Empty {}
message Message {
  string author = 1;
  string text = 2;
  google.protobuf.Timestamp send_time = 3;
}
message SendResponse {
  bool success = 1;
  string id = 2;
  google.protobuf.Timestamp server_time = 3;
}
service Messaging {
  rpc MessageStream (Empty) returns (stream Message);
  rpc SendMessage (Message) returns (SendResponse);
}